velox/dwio/parquet/tests/ParquetTpchTest.cpp

```diff
@@ -36,68 +34,93 @@ namespace fs = std::filesystem;
 namespace fs = std::experimental::filesystem;
 #endif
 
+#include <map>
+#include <vector>
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::exec::test;
+
+using TableColumnsMap = std::map<std::string, std::vector<std::string>>;
+
 class ParquetTpchTest : public testing::Test {
  protected:
   // Setup a DuckDB instance for the entire suite and load TPC-H data with scale
   // factor 0.01.
-  static void SetUpTestCase() {
+  static void SetUpTestSuite() {
     if (duckDb_ == nullptr) {
       duckDb_ = std::make_shared<DuckDbQueryRunner>();
       constexpr double kTpchScaleFactor = 0.01;
       duckDb_->initializeTpch(kTpchScaleFactor);
     }
     functions::prestosql::registerAllScalarFunctions();
-  }
-
-  void SetUp() override {
+    facebook::velox::parse::registerTypeResolver();
     filesystems::registerLocalFileSystem();
     parquet::registerParquetReaderFactory();
+    tempDirectory_ = exec::test::TempDirectoryPath::create();
+
     auto hiveConnector =
         connector::getConnectorFactory(
             connector::hive::HiveConnectorFactory::kHiveConnectorName)
             ->newConnector(kHiveConnectorId, nullptr);
     connector::registerConnector(hiveConnector);
-    tempDirectory_ = exec::test::TempDirectoryPath::create();
-    lineitemType_ =
-        ROW({"orderkey",
-             "partkey",
-             "suppkey",
-             "linenumber",
-             "quantity",
-             "extendedprice",
-             "discount",
-             "tax",
-             "returnflag",
-             "linestatus",
-             "shipdate",
-             "commitdate",
-             "receiptdate",
-             "shipinstruct",
-             "shipmode",
-             "comment"},
-            {BIGINT(),
-             BIGINT(),
-             BIGINT(),
-             BIGINT(),
-             DOUBLE(),
-             DOUBLE(),
-             DOUBLE(),
-             DOUBLE(),
-             VARCHAR(),
-             VARCHAR(),
-             DATE(),
-             DATE(),
-             DATE(),
-             VARCHAR(),
-             VARCHAR(),
-             VARCHAR()});
+
+    // Convert decimal columns to double.
+    const std::string lineitemSQL = R"(
```

Comment:
nit: consider extracting helper functions to create different tables, e.g. createLineitem, createOrders and createCustomer, for readability:

```
parquetFilesPaths_["lineitem"] = createLineitem();
parquetFilesPaths_["orders"] = createOrders();
parquetFilesPaths_["customer"] = createCustomer();
```

Reply:
Unless there is a strong inclination towards such arrangement of code, I honestly don't see the need to pull out these string literals and wrap them with functions unless we plan on using these strings somewhere else.

Reply:
This method in its current form is very long and hard to read. As we add more tables it will get longer and longer and harder and harder to read.